File: d1/solution_one.py
INITIAL_VAL = 50
CIRCLE_SIZE = 100
INPUT_PATH = 'input.txt'


def read_input(input_path=INPUT_PATH):
    with open(input_path, 'r') as file:
        data = [line.strip() for line in file.readlines()]
        int_data = [int(line[1:]) if line[0] == 'R' else -
                    int(line[1:]) for line in data]
    return int_data


def count_zeros(int_list, initial_value=INITIAL_VAL, circle_size=CIRCLE_SIZE):
    k = initial_value
    z = int(k == 0)

    for val in int_list:
        k = (k + val) % circle_size
        z += (k % circle_size == 0)
    return z


def main():
    input_data = read_input()
    print(count_zeros(input_data))


if __name__ == "__main__":
    main()

File: d1/solution_one.jl
const INITIAL_VAL = 50

function read_input(path::String="input.txt")::Vector{Int}
    lines = Int[]
    open(path, "r") do file
        for line in eachline(file)
            if line[1] == 'R' # always single quotation for char
                push!(lines, parse(Int, line[2:end]))
            else
                push!(lines, -parse(Int, line[2:end]))
            end
        end
    end

    return lines
end

function count_zeros(input::Vector{Int}, initial_val::Int=INITIAL_VAL)::Int
    counter = 0
    if initial_val == 0
        counter += 1
    end

    for value in input
        initial_val += value
        if initial_val % 100 == 0
            counter += 1
        end
    end

    return counter
end

# Main function
function main()
    input_list = read_input()
    ans = count_zeros(input_list)
    println(ans)
end


# Run main if script is executed directly
if abspath(PROGRAM_FILE) == @__FILE__
    main()
end
